data/transactions/logic/assembler.go

```diff
@@ -1142,19 +1148,98 @@ func AssembleStringWithVersion(text string, version uint64) ([]byte, error) {
 	return program, err
 }
 
-// AssembleStringWithVersionEx takes an entire program in a string and assembles it to bytecode using the assembler version specified
-// It also returns PC to source line mapping
+// AssembleStringWithVersionEx takes an entire program in a string and assembles it to bytecode
+// using the assembler version specified.
+// If version is zero it uses #pragma version or fallbacks to AssemblerDefaultVersion.
+// It also returns PC to source line mapping.
 func AssembleStringWithVersionEx(text string, version uint64) ([]byte, map[int]int, error) {
 	sr := strings.NewReader(text)
+	ps := PragmaStream{}
```

Comment:
I'm surprised this didn't wind up in the normal line handling inside OpStream.Assemble()
If OpStream.Version is zero at the time of the first real opcode, that could be an error, but it could still be set from the inside while processing early lines.

Reply:
It was there initially but in this case the parsing loop becomes too heavy - `version` can be only on first line, and before processing anything else an actual version needs to be determined. Other possible future pragmas may occur on other lines that adds more conditions.